newIDE/app/src/EventsSheet/InstructionEditor/EventTextDialog.js

```diff
@@ -0,0 +1,240 @@
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  textValue: string,
+  textColor: RGBColor,
+  backgroundColor: RGBColor,
+|};
+
+export default class EventTextDialog extends React.Component<Props, State> {
+  _applyChangesOnEvent = () => {
+    const { event } = this.props;
+    const { textValue, textColor, backgroundColor } = this.state;
+    const eventType = event.getType();
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      //Text value
+      gd.asCommentEvent(event).setComment(textValue);
+
+      //Text color
+      gd.asCommentEvent(event).setTextColor(
+        parseInt(textColor.r),
+        parseInt(textColor.g),
+        parseInt(textColor.b)
+      );
+      //Background color
+      gd.asCommentEvent(event).setBackgroundColor(
+        parseInt(backgroundColor.r),
+        parseInt(backgroundColor.g),
+        parseInt(backgroundColor.b)
+      );
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      //Text value
+      gd.asGroupEvent(event).setName(textValue);
+
+      //Text color for group not supported in Core, instead GroupEvent.js handle this
+      //Background color
+      gd.asGroupEvent(event).setBackgroundColor(
+        parseInt(backgroundColor.r),
+        parseInt(backgroundColor.g),
+        parseInt(backgroundColor.b)
+      );
+      return;
+    }
+  };
+
+  _getInitialStateFromEvent = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let text: string, textColors: RGBColor, backgroundColors: RGBColor;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      textColors = {
+        r: commentEvent.getTextColorRed(),
+        g: commentEvent.getTextColorGreen(),
+        b: commentEvent.getTextColorBlue(),
+      };
+
+      backgroundColors = {
+        r: commentEvent.getBackgroundColorRed(),
+        g: commentEvent.getBackgroundColorGreen(),
+        b: commentEvent.getBackgroundColorBlue(),
+      };
+
+      text = gd.asCommentEvent(event).getComment();
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      var groupEvent = gd.asGroupEvent(event);
+      const r = groupEvent.getBackgroundColorR(),
+        g = groupEvent.getBackgroundColorG(),
+        b = groupEvent.getBackgroundColorB();
+
+      const white = {
+        r: 255,
+        g: 255,
+        b: 255,
+      };
+
+      const black = {
+        r: 0,
+        g: 0,
+        b: 0,
+      };
+
+      textColors = (r + g + b) / 3 > 200 ? black : white; //Because text color is not supported by Core
+
+      backgroundColors = {
+        r: groupEvent.getBackgroundColorR(),
+        g: groupEvent.getBackgroundColorG(),
+        b: groupEvent.getBackgroundColorB(),
+      };
+
+      text = gd.asGroupEvent(event).getName();
+    } else {
+      console.error(
+        'Dialog was opened for an unsupported event type: ' + eventType
+      );
+    }
```

Comment:
Flow will not let you make risky things: it has seen that you declared text, textColors, backgroundColors but have not set them a value there. Always make sure that whatever the "branch" of the if/else is being used, your variables are initialised.

This is what I explained in a previous comment. Note that you could also set the variables in the "else" (`text = ''; textColor = { ... }`) but I prefer to initialise your variables as soon as possible.